internal/analyzer/ktn/tests/quality/deadtablebranch_internal_test.go

```
@@ -0,0 +1,127 @@
+	}
+}
+
+// Test_branchCondition_BinaryExpr covers the BinaryExpr dispatch
+// independently because parseFuncBodyDeadtable wraps statements, while
+// BinaryExpr is naturally an expression — we synthesise it directly.
+func Test_branchCondition_BinaryExpr(t *testing.T) {
+	expr, err := parser.ParseExpr("a == b")
+	if err != nil {
+		t.Fatalf("parse: %v", err)
+	}
+	got := branchCondition(expr.(ast.Node))
```

golangci-lint (2.12.1)

[error] right hand must be only type assertion

(forcetypeassert)